lib/ttdrest/concerns/base.rb

```diff
@@ -20,16 +25,67 @@ def authenticate(options = {})
       def check_response(response)
         if response.code.eql?("403")
           raise AuthorizationFailedError
+        elsif retryable_http_error?(response)
+          raise RecoverableHttpError
+        end
+      end
+
+      def retryable_http_error?(response)
+        response && (
+          response.code == 429 || (500..599).include?(response.code)
+        )
+      end
+
+      def parse_header_retry(response)
+        return nil unless response['Retry-After']
+
+        if response['Retry-After'] =~ /^\d+$/
+          sleep(response['Retry-After'].to_i)
```

Comment:
questioning the api doc, is this retry after utc time or a number of seconds, this is a number of seconds.

Reply:
Yes the docs say a # of seconds backoff, or a date string.